### DISCUSSION TITLE
Add Let's Encrypt Rate Limit Warning

### DISCUSSION THREAD

**Discussion by acetousk** [nextcloud/all-in-one#515] [Ideas] [answered]
#### What I noticed

I just spent a couple hours trying to figure out why when I loaded my domain name, I got this error:
![image](https://user-images.githubusercontent.com/29029373/164816529-53820a10-8679-4058-8055-3655e3d387d6.png)
This was due to an error that I got when running
```bash
user@computer$ sudo docker logs nextcloud-aio-apache
```
At the bottom it said:
```json
{"level":"error","ts":1650668233.3344505,"logger":"tls.obtain","msg":"will retry","error":"[********.com] Obtain: [********.com] creating new order: attempt 1: https://acme-v02.api.letsencrypt.org/acme/new-order: HTTP 429 urn:ietf:params:acme:error:rateLimited - Error creating new order :: too many certificates (5) already issued for this exact set of domains in the last 168 hours: ********.com: see https://letsencrypt.org/docs/rate-limits/ (ca=https://acme-v02.api.letsencrypt.org/directory)","attempt":4,"retrying_in":300,"elapsed":303.433983656,"max_duration":2592000}
```
This is because I had tried several different times to get a certificate from Let's Encrypt and was being rate limited on the domain (see: https://letsencrypt.org/docs/rate-limits/).

I tried a different domain after resetting the data by first stopping containers on the aio interface, then:
```bash
sudo docker container stop nextcloud-aio-mastercontainer
```
Warning: Only run the following commands if you don't have data that you need on containers that you have stopped.
```bash
sudo docker container prune && sudo docker volume prune
```

Once restarting it and running the aio again with a domain that I hadn't been using and it worked fine. 

#### What to do
Mostly I'm just letting the community know what my problem is in hopes of helping solve someone elses problem in the future. 

Maybe at some point, have an error if the acme certification process responds with an error so that people can know what is going on.

**Comment by szaimen** [comment, marked as answer]
This is a good idea in general but unfortunately not really possible without much effort as we would need to filter for this specific log inside the apache container logs and react in the aio interface based on this...
And getting the logs every time the page loads is pretty expensive...
